### PR TITLE
Fixes #555: Moved download_sample to JobViewSet action

### LINKED ISSUE #555: [views] Move `download_sample` to an `@action("download-sample")` inside `JobViewSet`.

### PR DESCRIPTION
**Pull request by m0mosenpai**
## Related issues
#555 

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] The tests gave 0 errors.
- [X] `Black` gave 0 errors.
- [X] `Flake` gave 0 errors.

### CODE REVIEW

**Comment by m0mosenpai** [comment]
@Eshaan7 
Is this purely for refactoring reasons or does it have any particular benefit?

File: api_app/api.py [hunk 1]
```diff
@@ -376,19 +335,52 @@ def kill(self, request, pk=None):
...
+        data_received = request.query_params
+        logger.info(f"Get binary by Job ID. Data received {data_received}")
+        if "job_id" not in data_received:
+            return Response({"error": "821"}, status=status.HTTP_400_BAD_REQUEST)
+
+        # get job object or raise 404
+        try:
+            job = self.get_object()
+        except job.DoesNotExist:
+            raise NotFound()
```
Comment (Eshaan7) [inline api_app/api.py:369]: All this can be replaced by `self.get_object()` as done on line 339-340.

**Comment by Eshaan7** [comment]
- [ ] fix `api.py` 
- [ ] fix URL in `test_api`.

File: api_app/api.py (hunk 1 above)
Comment (Eshaan7) [reply api_app/api.py:369]:
```suggestion
# get job object
job = self.get_object()
```

**Comment by Eshaan7** [comment]
Also: merge this PR with latest changes of `develop-2` so the tests run properly.

**Review by Eshaan7 — APPROVED** [review]

**Pull request merged**